third_party/move/move-core/types/src/resolver.rs

```diff
@@ -40,12 +41,23 @@ pub fn resource_size(resource: &Option<Vec<u8>>) -> usize {
 ///                       are always structurally valid)
 ///                    - storage encounters internal error
 pub trait ResourceResolver {
-    fn get_resource_with_metadata(
+    // Note:
+    //
+    // This methods takes an optional layout argument in order to pass it to
+    // the adapter so that aggregator liftings can be handled.
+    // If it is None:
```

Comment:
Why not use get_resource_with_metadata in that case? i.e. why not always provide a layout here?

Reply:
Same as for tables: motivation was to reuse the flow, but I agree that we can just pass the layout. It is also nicer because it is what we need for Move values :)